src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -73,58 +83,131 @@ public void storeMessage(Message message) {
-    Query query =
-        new Query("Message")
-            .setFilter(directMessages)
-            .addSort("timestamp", SortDirection.ASCENDING);
+  public List<Message> getAllMessages() {
+    Query query = createPublicMessageQuery();
     PreparedQuery results = datastore.prepare(query);
+    return prepareQueryToMesssages(results);
+  }
 
-    for (Entity entity : results.asIterable()) {
-      String idString = entity.getKey().getName();
-      UUID id = UUID.fromString(idString);
-      String text = getStringProperty(entity, "text").orElse("");
-      long timestamp = (long) entity.getProperty("timestamp");
-      String sender = (String) entity.getProperty("user");
-      String receiver = (String) entity.getProperty("recipient");
-      Message message = new Message(id, sender, text, timestamp, receiver);
-      messages.add(message);
+  private List<Message> prepareQueryToMesssages(PreparedQuery query) {
+    List<Message> results = new ArrayList<>();
+    for (Entity entity : query.asIterable()) {
+      results.add(entityToMessage(entity));
     }
+    return results;
+  }
 
-    return messages;
+  private Message entityToMessage(Entity entity) {
+    String idString = entity.getKey().getName();
+    UUID id = UUID.fromString(idString);
+    String text = getStringProperty(entity, "text").orElse("");
+    long timestamp = (long) entity.getProperty("timestamp");
+    String sender = (String) entity.getProperty("user");
+    String receiver = (String) entity.getProperty("recipient");
+    return new Message(id, sender, text, timestamp, receiver);
+  }
+
+  private Query createPublicMessageQuery() {
+    return new Query("Message").addSort("timestamp", SortDirection.ASCENDING);
   }
 
-  /** Returns the total number of messages for all users. */
   public int getTotalMessageCount() {
     Query query = new Query("Message");
     PreparedQuery results = datastore.prepare(query);
     return results.countEntities(FetchOptions.Builder.withDefaults());
   }
 
+  /**
+   * This is to create an user form data to put in the datastore Going with option 2 highlighted in
+   * the Meeting Notes
+   *
+   * @throws EntityNotFoundException
+   */
+  public void storeUserTeaData(
+     Map<String, Long> incomingUserTeaData, String username, String date) {
+
+    Entity datastoreUser;
+    String keyName = date + ":" + username;
+    Key usernameKey = KeyFactory.createKey("UserTeaData", keyName);
+
+    try {
+      datastoreUser = datastore.get(usernameKey);
+      datastoreUser.setProperty("username", datastoreUser.getProperty("username"));
+      datastoreUser.setProperty("date", datastoreUser.getProperty("date"));
+
+      EmbeddedEntity teaMap = (EmbeddedEntity) datastoreUser.getProperty("teaData");
+
+      for (String key : teaMap.getProperties().keySet()) {
+        Long value1 = incomingUserTeaData.get(key);
+        Long value2 = value1 + (Long) teaMap.getProperty(key);
```

Comment:
Why are you assuming that it is additive? That doesn't seem to reflect the nature of the form. I would assume that the data should simply be overwritten with the new values.